pilot/pkg/features/pilot.go

```diff
@@ -447,6 +447,9 @@ var (
 
 	EnableEnvoyFilterMetrics = env.RegisterBoolVar("PILOT_ENVOY_FILTER_STATS", false,
 		"If true, Pilot will collect metrics for envoy filter operations.").Get()
+
+	MulticlusterHeadlessEnabled = env.RegisterBoolVar("ENABLE_MULTICLUSTER_HEADLESS", false,
```

Comment:
Maybe this flag is redundant and can be enabled by default

Reply:
I agree that it should eventually be default. John brought up a concern, which I agree with, that some user may have headless services that behaved in a cluster-local way before, and to change that might be surprising. 

I'm hoping to push this with the flag first, then turn it on by default with an upgrade note. This also lets us backport the change.

Reply:
Maybe it's ok for 1.11, but let's merge this first, backport then discuss turning it on by default/removing the flag from the future codebase.

Reply:
SGTM